homeassistant/components/solax/__init__.py

```diff
@@ -21,19 +37,29 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             entry.data[CONF_PORT],
             entry.data[CONF_PASSWORD],
         )
-        await api.get_data()
     except Exception as err:
         raise ConfigEntryNotReady from err
 
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = api
+    async def _async_update() -> InverterResponse:
+        try:
+            return await api.get_data()
+        except InverterError as err:
+            raise UpdateFailed from err
+
+    coordinator = DataUpdateCoordinator[InverterResponse](
```

Comment:
I would love to see this one in `coordinator.py`

Reply:
moved in cc207e92a02359dd9b485d7cd6004297beaff97a